metrics/sinks/stackdriver/stackdriver.go

```diff
@@ -213,15 +217,92 @@ func (sink *StackdriverSink) ExportData(dataBatch *core.DataBatch) {
 				req.TimeSeries = append(req.TimeSeries, point)
 			}
 			if len(req.TimeSeries) >= maxTimeseriesPerRequest {
-				sink.requestQueue <- req
+				requests = append(requests, req)
 				req = getReq()
 			}
 		}
 	}
 
 	if len(req.TimeSeries) > 0 {
-		sink.requestQueue <- req
+		requests = append(requests, req)
 	}
+
+	go sink.sendRequests(requests)
+}
+
+func (sink *StackdriverSink) sendRequests(requests []*sd_api.CreateTimeSeriesRequest) {
+	// Each worker can handle at least batchExportTimeout/sdRequestLatencySec requests within the specified period.
+	// 5 extra workers just in case.
+	workers := 5 + len(requests)/(sink.batchExportTimeoutSec/sdRequestLatencySec)
+	requestQueue := make(chan *sd_api.CreateTimeSeriesRequest)
+	// Close the channel in order to cancel exporting routines.
+	defer close(requestQueue)
+
+	// Launch Go routines responsible for sending requests
+	for i := 0; i < workers; i++ {
+		go sink.requestSender(requestQueue)
+	}
+
+	timeoutTime := time.Now().Add(time.Duration(sink.batchExportTimeoutSec) * time.Second)
+
+forloop:
+	for i := range requests {
+		now := time.Now()
+		select {
+		case requestQueue <- requests[i]:
+			// yet another request added to queue
+		case <-time.After(timeoutTime.Sub(now)):
```

Comment:
This is misleading. You can add all requests to the queue but it doesn't mean that the operation completed.